### PR TITLE
Add rebase check to CI

### PR DESCRIPTION
**Pull request by HenryNguyen5**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -340,47 +348,73 @@ workflows:
   version: 2
   test:
     jobs:
+      - check_if_rebase_needed:
+          filters: # all branches, and /^v../ tags for build-publish...
+            tags:
+              only: /^v.*/
       - go-sqlite:
+          requires:
+            - check_if_rebase_needed
```
Comment (felder-cl) [inline .circleci/config.yml]: Why does the rebase test stop other tests from running on failure?

**Comment by j16r** [comment]
@HenryNguyen5 rebase

File: .circleci/config.yml (hunk 1 above)
Comment (HenryNguyen5) [reply .circleci/config.yml]: I was thinking that the tests would be stale, so there might not be any point in running them if you havent rebased. Would you rather them run anyway?

Comment (HenryNguyen5) [reply .circleci/config.yml]: Reverted this change, i see the value in it now

**Pull request merged**